Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,138 @@
+// DependencyListNode.cpp
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "DependencyListNode.h"
+
+#include "Tools/FBuild/FBuildCore/FBuild.h"
+#include "Tools/FBuild/FBuildCore/FLog.h"
+#include "Tools/FBuild/FBuildCore/BFF/Functions/Function.h"
+#include "Tools/FBuild/FBuildCore/Graph/NodeGraph.h"
+
+#include "Core/Containers/Array.h"
+#include "Core/Env/ErrorFormat.h"
+#include "Core/FileIO/FileIO.h"
+#include "Core/FileIO/FileStream.h"
+#include "Core/FileIO/PathUtils.h"
+#include "Core/Strings/AStackString.h"
+
+// REFLECTION
+//------------------------------------------------------------------------------
+REFLECT_NODE_BEGIN( DependencyListNode, Node, MetaName("Dest") + MetaFile() )
+    REFLECT(        m_Source,                   "Source",                   MetaFile() )
```

Comment:
This will allow "Source" to point to a file (like a library) directly or via an alias, but won't allow you to point to an ObjectList or certain other types of nodes.

I think you might want MetaAllowNonFile here.